vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helpers/AuthenticationResponseResultFactory.kt

```diff
@@ -0,0 +1,175 @@
+package at.asitplus.wallet.lib.oidc.helpers
+
+import at.asitplus.KmmResult
+import at.asitplus.KmmResult.Companion.wrap
+import at.asitplus.crypto.datatypes.jws.JsonWebKey
+import at.asitplus.crypto.datatypes.jws.JsonWebKeySet
+import at.asitplus.crypto.datatypes.jws.JweAlgorithm
+import at.asitplus.crypto.datatypes.jws.JweEncryption
+import at.asitplus.crypto.datatypes.jws.JweHeader
+import at.asitplus.crypto.datatypes.jws.toJsonWebKey
+import at.asitplus.wallet.lib.jws.JwsService
+import at.asitplus.wallet.lib.oidc.AuthenticationRequestSource
+import at.asitplus.wallet.lib.oidc.AuthenticationResponseParameters
+import at.asitplus.wallet.lib.oidc.AuthenticationResponseResult
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import at.asitplus.wallet.lib.oidc.RelyingPartyMetadata
+import at.asitplus.wallet.lib.oidvci.OAuth2Exception
+import at.asitplus.wallet.lib.oidvci.encodeToParameters
+import at.asitplus.wallet.lib.oidvci.formUrlEncode
+import io.github.aakira.napier.Napier
+import io.ktor.http.URLBuilder
+import kotlin.random.Random
+
+
+class AuthenticationResponseResultFactory(
+    val jwsService: JwsService,
+) {
+    suspend fun createAuthenticationResponseResult(
+        responsePreparationState: AuthenticationResponsePreparationState,
+        responseParameters: AuthenticationResponseParameters,
+    ): KmmResult<AuthenticationResponseResult> {
+        return when (responsePreparationState.responseModeParameters) {
```

Comment:
Please convert to expression body, and use `= runCatching { ... throw Exception() ...  }.wrap()` to make it more readable